### ISSUE TITLE
The date filter is somewhat broken

### ISSUE THREAD

**Issue by nesk** [jarednova/timber#545]
Say I have a timestamp equal to `1429488000`. If I write this in a template:

```twig
{{ 1429488000|date('c') }}
```

I get __8000-05-22T14:29:48+00:00__, which is not what I was expecting, the correct value is __2015-04-20T00:00:00+00:00__.

This error is due to this line: https://github.com/jarednova/timber/blob/master/lib/timber-twig.php#L246

Before creating a pull request to replace this line by `$timestamp = intval($date);`, I wanted to know if the current behaviour was expected? If so, why? Is there any reason to do this?

**Comment by xavivars** [comment]
Probably a better aproach is create another "if" there so, if `$date` is an int, then apply `intval`, but not doing it when is a string (as, probably, it expects a "well formated date as a string")

**Comment by bigmike7801** [comment]
Is there a solution or a work around to this bug?

**Comment by bigmike7801** [comment]
This is what I did:

        /**
         *
         *
         * @param string  $date
         * @param string  $format (optional)
         * @return string
         */
        function intl_date( $date, $format = null ) {
        	if ( $format === null ) {
        		$format = get_option( 'date_format' );
        	}
        	
        	if(is_int($date)){
        		$timestamp = $date; 
        	}elseif ( $date instanceof DateTime ) {
        		$timestamp = $date->getTimestamp();
        	} else {
        		$timestamp = strtotime( $date );
        	}
        	
        	return date_i18n( $format, $timestamp );
        }

**Comment by nesk** [comment]
I just redeclared the `date` filter:

```php
add_action('twig_apply_filters', function($twig) {

    $twig->addFilter('date', new Twig_Filter_Function(function($date, $format = null) {
        if ( $format === null ) {
            $format = get_option( 'date_format' );
        }

        if ($date instanceof DateTime) {
            $timestamp = $date->getTimestamp();
        } else if ($date != 'now') {
            $timestamp = intval($date);
        } else {
            $timestamp = strtotime($date);
        }

        return date_i18n($format, $timestamp);
    }));

    return $twig;

});
```

**Comment by jarednova** [comment]
Thanks to @xavivars for the fix in #715!
